main.go

```diff
@@ -118,71 +146,56 @@ func main() {
 	}
 	stepconf.Print(config)
 
-	if config.NpmVersion == "" {
+	workdir, err := pathutil.AbsPath(config.Workdir)
+	if err != nil {
+		failf("error normalizing workdir path: %s", err)
+	}
+	if _, err := os.Stat(workdir); os.IsNotExist(err) {
+		failf("specified path `%s` does not exist", workdir)
+	}
+
+	toInstall := false
+	userDefined := config.NpmVersion
+	toSet := userDefined
+
+	if toSet == "" {
 		fmt.Println()
 		log.Infof("Autodetecting npm version")
-
 		log.Printf("Checking package.json for npm version")
-		ver, err := getNpmVersionFromPackageJSON()
-		if err != nil {
-			failf("error processing package.json: %s", err)
-		}
+		toSet = readFromPackageJSON(workdir)
+	}
 
-		if ver == "" {
-			log.Warnf("No npm version found in package.json")
-
-			log.Printf("Locating system installed npm")
-			path, err := exec.LookPath("npm")
-			if err != nil {
-				log.Warnf("npm not found on PATH")
-
-				log.Printf("Installing latest npm")
-				ver = "latest"
-
-				out, err := runAndLog(createInstallNpmCommand(runtime.GOOS))
-				if err != nil {
-					log.Errorf(out)
-					failf("Error installing npm: %s", err)
-				}
-
-				log.Printf(out)
-			} else {
-				log.Printf("npm found at %s", path)
-
-				cmd := command.New("npm", "--version")
-				out, err := runAndLog(cmd)
-				if err != nil {
-					log.Warnf("Error getting installed npm version: %s", err)
-				}
-				ver = out
-			}
-		} else {
-			fmt.Println(ver)
-			fmt.Println()
-			log.Infof("Setting npm version to %s", ver)
-
-			out, err := setNpmVersion(ver)
-			if err != nil {
-				log.Errorf(out)
-				failf("Error setting npm version to %s: %s", ver, err)
-			}
+	if toSet == "" {
+		log.Warnf("No npm version found in package.json")
+		log.Printf("Locating system installed npm")
+		toSet = systemDefined()
+		if toSet == "" {
+			toSet = "latest"
```

Comment:
leave this empty as later you will use this var to determine if there is an npm version to set or not, but toInstall true will lead to the installation of latest npm.